app/src/main/java/by/academy/jc/ht/skopinov/ht13/MyClass.java

```diff
@@ -0,0 +1,55 @@
+package by.academy.jc.ht.skopinov.ht13;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class MyClass {
+
+  String[] names = {"Leyli", "Nataliya", "Alexey", "Artut", "Ekaterina", "Vladislav", "Iliya", "Valeriya", "Dmitry", "Timofey", "Ivan", "Petr", "Sergey", "Maxim", "Pavel"};
+  String[] surnames = {"Akbarova", "Dubovtsova", "Zhabko", "Zabrodskiy", "Zaykovskaya", "Klimahovich", "Lukashevich", "Sergeychik", "Skopinov", "Shorohov", "Ivanov", "Petrov", "Smirnov", "Sidorov", "Pavlov"};
+  int amountOfPersons = 100;
+
+  public List<Person> creatingPersons(String[] names, String[] surnames, int amountOfPersons) {
+    this.names = names;
+    this.surnames = surnames;
+    this.amountOfPersons = amountOfPersons;
+    List<Person> group = new ArrayList<>();
+    for (int i = 0; i < amountOfPersons; i++) {
+      group.add(new Person(names[(int) (Math.random() * names.length)],
+          surnames[(int) (Math.random() * surnames.length)],
+          (int) (Math.random() * 16) + 15));
+    }
+    return group;
+  }
+
+  public Set<Character> mySurnameLetters(String surname){
+    Set<Character> letters = new HashSet<>();
+    char[] chars = surname.toLowerCase().toCharArray();
+    for (char c : chars) {
+      letters.add(c);
+    }
+    return letters;
+  }
+
+  public Set<String> customCollector(List<Person> group, Set<Character> mySurnameLetters) {
```

Comment:
Custom collector should implements https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Collector.html right?

If it's a factory method, it should be static and called like `createCustomCollector()`